### PR TITLE
CMR-3858 Update Catalog Item ACL endpoints to accept concept_id

### PR DESCRIPTION
**Pull request by daniel-zamora**
This pull request creates a new migration to fix corrupted catalog item ACLs.

### CODE REVIEW

File: metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
   (when-let [entry-titles (get collection-identifier :entry-titles)]
-      (let [provider (-> (h/get-provider provider-id)
-                         providers/dbresult->provider)
-            t (h/get-provider-collection-tablename provider)]
-        (for [entry-title entry-titles
-              result (h/query [(format "select concept_id from metadata_db.%s where
-                                        entry_title = ?" t)
-                               entry-title])]
-          (:concept_id result)))))
+    (let [provider (-> (h/get-provider provider-id)
```
Comment (oubiwann) [inline metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj]: Quick note on the threading: kibit will fail that unless there are three levels getting passed. Quick fix:
```clj
(-> provider-id
    (h/get-provider)
    providers/dbresult->provider)
```

File: metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj [hunk 2]
```diff
@@ -11,14 +11,14 @@
...
+                       providers/dbresult->provider)
+          t (h/get-provider-collection-tablename provider)]
+      (for [entry-title entry-titles
+            result (j/query (config/db) [(format "select concept_id from metadata_db.%s where
```
Comment (oubiwann) [inline metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj]: The wrong namespace's query was getting called? Completely optional, depending on how pressing this is and how you guys usually handle bug-fixes in tests, but I'd be +1 for adding a test that duplicated the error, but with the fix in place, passes the test. Maybe check the format? Not sure how `h/query` and `j/query` differ ...

File: metadata-db-app/src/migrations/054_fix_catalog_acl_concept_ids.clj [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  []
+  (println "migrations.054-fix-catalog-acl-concept-ids up...")
+  (doseq [result (h/query "select * from cmr_acls where acl_identity like 'catalog-item%'")]
+    (println result)
+    (let [{:keys [id metadata deleted]} result
+          deleted (= 1 (long deleted))
+          ;;Remove bad parts of metadata string
+          metadata (-> metadata
+                       (util/gzip-blob->string)
+                       (string/replace #"Executing.*\n" "")
```
Comment (oubiwann) [inline metadata-db-app/src/migrations/054_fix_catalog_acl_concept_ids.clj]: Would really like to see some unit tests that take corrupted sample data and show this cleaning it up ...

File: metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj (hunk 2 above)
Comment (daniel-zamora) [inline metadata-db-app/src/migrations/052_add_coll_ids_catalog_acls.clj]: For some reason the println in h/query was getting into the concept-ids vector, not sure how.

**Pull request merged**